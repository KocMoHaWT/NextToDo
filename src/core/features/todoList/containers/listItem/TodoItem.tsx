import React from 'react';
import classnames from 'classnames';
import styles from './todoItem.component.scss';
import { ToDo } from '../../../../shared/interfaces/toDo';
import { CheckBox } from '../../../../shared/coreUi/checkBox/CheckBox';
import { Button } from '../../../../shared/coreUi/button/Button';

interface TodoItem {
  todo: ToDo;
  onDelete?: () => void;
  onComplete?: () => void;
}

export const TodoItem: React.FunctionComponent<TodoItem> = ({ onComplete, onDelete, todo }) => {
  const todoClass = classnames(styles.listTodo, { [styles.completed]: todo.completed });
  return (
    <li className={todoClass}>
      {todo.info}
      <div className={styles.buttons}>
        <CheckBox name="complete todo" onChange={onComplete} defaultValue={todo.completed} />
        <Button type="button" borderLess onClick={onDelete} className={styles.deleteBtn}>
          Ã—
        </Button>
      </div>
    </li>
  );
};
