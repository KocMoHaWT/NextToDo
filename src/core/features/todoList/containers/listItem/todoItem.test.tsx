import React from 'react';
import { create } from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import { TodoItem } from './TodoItem';
import '@testing-library/jest-dom/extend-expect';

const testLabel = 'djf;lakjdf';

describe('Button', () => {
  afterEach(cleanup);

  it('should be text', () => {
    const { getByText } = render(<TodoItem todo={{ id: 0, info: testLabel, completed: false }} />);
    const element = getByText(testLabel);
    expect(element).toBeInTheDocument();
  });

  it('should be completed', () => {
    const { getByText } = render(<TodoItem todo={{ id: 0, info: testLabel, completed: true }} />);
    const element = getByText(testLabel);
    expect(element).toHaveClass('completed');
  });

  it('todo item renders correctly', () => {
    const tree = create(<TodoItem todo={{ id: 0, info: testLabel, completed: false }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
