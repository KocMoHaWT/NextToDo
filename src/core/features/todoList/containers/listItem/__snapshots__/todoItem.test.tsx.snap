// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button todo item renders correctly 1`] = `
<li
  className="listTodo"
>
  djf;lakjdf
  <div
    className="buttons"
  >
    <label
      className="container container"
    >
      <span
        className="label"
      >
        
      </span>
      <input
        className="checkBox"
        defaultChecked={false}
        name="complete todo"
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </label>
    <button
      className="button unborder deleteBtn"
      type="button"
    >
      Ã—
    </button>
  </div>
</li>
`;
